### PR TITLE
Sort archs and make the generate predictable

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Sorry for the noise, I didn't pay attention to the order of the archs. It's now predictable so this shouldn't happen again.

This PR adds the "general" intrinsics (not bound to any arch).

Last part of https://github.com/rust-lang/rustc_codegen_gcc/pull/151.

### CODE REVIEW

File: src/intrinsic/archs.rs [hunk 1]
```diff
@@ -2935,31 +1508,1475 @@ match name {
...
+    "llvm.objectsize.i32.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.i64.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.i8.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v16i8.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v2i64.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v4i32.p0i8" => "__builtin_object_size",
+    "llvm.objectsize.v8i16.p0i8" => "__builtin_object_size",
+    "llvm.stackrestore" => "__builtin_stack_restore",
+    "llvm.stacksave" => "__builtin_stack_save",
+    "llvm.trap" => "__builtin_trap",
```
Comment (bjorn3) [inline src/intrinsic/archs.rs:2980]: I don't think any of these are used by stdarch or any other part of the standard library.

Comment (bjorn3) [reply src/intrinsic/archs.rs:2980]: It is only used inside cg_llvm as implementation for the `abort` builder function. cg_gcc directly calls the `abort` builtin function there.

Comment (antoyo) [reply src/intrinsic/archs.rs:2980]: `llvm.debugtrap` is needed by the `breakpoint` intrinsic.

Comment (bjorn3) [reply src/intrinsic/archs.rs:2980]: Wouldn't the implementation of said intrinsic directly call the right gcc builtin?

**Comment by GuillaumeGomez** [comment]
It seems like these functions shouldn't be added then. Is it fine to still have the sorting predictable?

**Comment by antoyo** [comment]
> It seems like these functions shouldn't be added then. Is it fine to still have the sorting predictable?

Indeed. We can add them later if needed.

Yes, please, go ahead for the sorting.

**Comment by GuillaumeGomez** [comment]
Done!

**Pull request merged**